src/main/java/edu/escuelaing/ieti/matchmaking/services/impl/UserServiceImpl.java

```diff
@@ -57,6 +64,34 @@ public User getUserByEmail(String email) throws EntityNotFoundException {
         return optionalUser.orElseThrow(() -> new EntityNotFoundException(User.class, "User email", email));
     }
 
+    @Override
+    public List<User> getUserFriendsById(String userId) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        List<User> friends = new ArrayList<>();
+        optionalUser.ifPresent(user -> {
```

Comment:
Instead of having a lambda here consider implementing a map operation in your stream to convert from Id to User object